src/Compilers/CSharp/Portable/Syntax/ArgumentSyntax.cs

```diff
@@ -1,17 +1,128 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+using System.Diagnostics;
+
 namespace Microsoft.CodeAnalysis.CSharp.Syntax
 {
     public partial class ArgumentSyntax
     {
+        /// <summary>
+        /// Get expression representing the value of the argument.
+        /// </summary>
+        public ExpressionSyntax Expression
+        {
+            get
+            {
+                if (RefOrOutKeyword.Kind() != SyntaxKind.OutKeyword)
+                {
+                    return (ExpressionSyntax)ExpressionOrDeclaration;
+                }
+
+                return ExpressionOrDeclaration as ExpressionSyntax;
+            }
+        }
+
+        public ArgumentSyntax WithExpression(ExpressionSyntax expression)
+        {
+            return this.Update(this.NameColon, this.RefOrOutKeyword, expression);
+        }
+
+        /// <summary>
+        /// Get declaration of an Out Variable for the argument.
+        /// </summary>
+        public VariableDeclarationSyntax VariableDeclaration
+        {
+            get
+            {
+                if (RefOrOutKeyword.Kind() != SyntaxKind.OutKeyword)
+                {
+                    return null;
+                }
+
+                return ExpressionOrDeclaration as VariableDeclarationSyntax;
+            }
+        }
+
+        /// <summary>
+        /// Get identifier for out variable declaration, or default(SyntaxToken).
+        /// </summary>
+        internal SyntaxToken Identifier
+        {
+            get
+            {
+                if (RefOrOutKeyword.Kind() == SyntaxKind.OutKeyword && ExpressionOrDeclaration.Kind() == SyntaxKind.VariableDeclaration)
+                {
+                    return ((VariableDeclarationSyntax)ExpressionOrDeclaration).Variables.First().Identifier;
+                }
+
+                return default(SyntaxToken);
+            }
+        }
+
+        /// <summary>
+        /// Get type syntax for out variable declaration, or null.
+        /// </summary>
+        internal TypeSyntax Type
+        {
+            get
+            {
+                if (RefOrOutKeyword.Kind() == SyntaxKind.OutKeyword && ExpressionOrDeclaration.Kind() == SyntaxKind.VariableDeclaration)
+                {
+                    return ((VariableDeclarationSyntax)ExpressionOrDeclaration).Type;
+                }
```

Comment:
Consider calling the `VariableDeclaration` property instead of the `if` condition used here and in `Identifier`, so there is one place where we determine `VariableDeclaration`.